crates/rpc/rpc-eth-api/src/helpers/call.rs

```diff
@@ -247,7 +247,12 @@ pub trait EthCall: Call + LoadPendingBlock {
                 state_context.unwrap_or_default();
             let transaction_index = transaction_index.unwrap_or_default();
 
-            let target_block = block_number.unwrap_or_default();
+            let block_number = block_number.unwrap_or_default();
+            let target_block: BlockId = LoadBlock::provider(self)
+                .block_hash_for_id(block_number)
+                .map_err(|_| EthApiError::HeaderNotFound(block_number))?
+                .ok_or_else(|| EthApiError::HeaderNotFound(block_number))?
+                .into();
```

Comment:
this does not really work for the pending block.

so we can only do this if this is not pending